### PR TITLE
feat: add token tag component (#13599)

### LINKED ISSUE #13599: UI Component - Token tag
This issue is to track the creation of the component: **Tag**

You can access the link at https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=2407%3A32482 .

Please have a chat with the design team before starting with the component, and don't forget to add sizing to the issue

Only light and dark mode needs to be implemented, the rest can be implemented separately.

### PR DESCRIPTION
**Pull request by J-Son89**
fixes #13599 

### Summary

Implements UI Component - Token tag

This implements and light and dark mode for the token tag component, it does not include the blurred versions of these as discussed with @cammellos and the solution for this needs to be discussed first.

#### Platforms

- Android. 
- iOS

#### Areas that maybe impacted
Test in quo 2 components

### Steps to test

- Open Status
- Open Options
- -Open Quo2 Preview

status: ready

### CODE REVIEW

File: src/quo2/components/token_tag.cljs
Comment (J-Son89) [inline src/quo2/components/token_tag.cljs]: perhaps there's a better way to get the purchase and checkmark icon sitting in the right place as per the designs 🤔

**Comment by Parveshdhull** [comment]
Hi @J-Son89, thank you for the contribution.
You need to rebase your branch against `develop` for fixing the build error.

File: src/quo2/components/token_tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/token_tag.cljs]: redundant else clause, can be replaced with when

Comment (J-Son89) [reply src/quo2/components/token_tag.cljs]: thanks, adjusted

File: src/quo2/components/token_tag.cljs
Comment (cammellos) [inline src/quo2/components/token_tag.cljs]: No need of commas between parameters

File: src/quo2/components/token_tag.cljs
Comment (cammellos) [inline src/quo2/components/token_tag.cljs]: Indentation is a bit off, we either leave a single space, or align values vertically

**Comment by Parveshdhull** [comment]
@J-Son89 linting failed, you can run `make lint` to see the errors. And then you can fix them manually or run `make lint-fix`

**Comment by Parveshdhull** [comment]
image
image

We should use same naming convention for all token icons. Maybe same as figma names, like
`token-logo-snt`, `token-logo-eth`, `token-logo-default`

File: src/quo2/components/token_tag.cljs
Comment (cammellos) [inline src/quo2/components/token_tag.cljs]: No need to use `(get-in themes [(theme/get-theme)])`, if not a nested map, you can use `get`

`(get themes (theme/get-theme))`

or (fun facts)

`(themes (theme/get-theme))`
`((theme/get-theme) themes))`

as both `keywords` and `maps` implement `IFn` (https://clojure.github.io/clojure/javadoc/clojure/lang/IFn.html ) and implement `get`.

`(get themes (theme/get-theme))` is the most readable :)

**Review by cammellos — COMMENTED** [review]
Looks great, thank you, I think just a few style issues & linting, great job!

File: src/quo2/components/token_tag.cljs
Comment (J-Son89) [reply src/quo2/components/token_tag.cljs]: awesome, thanks Andrea! :)

Comment (J-Son89) [reply src/quo2/components/token_tag.cljs]: updated

File: src/quo2/components/token_tag.cljs
Comment (J-Son89) [reply src/quo2/components/token_tag.cljs]: cool, I set it to a single space

File: src/quo2/components/token_tag.cljs
Comment (J-Son89) [reply src/quo2/components/token_tag.cljs]: updated

File: src/quo2/screens/main.cljs
Comment (Parveshdhull) [inline src/quo2/screens/main.cljs]: better to use `quo2-token-tag`, so that all screen names looks similar

**Comment by Parveshdhull** [comment]
Also result looks different when is required or is Purchasable is enabled,
Design:
image
PR:
File

**Comment by J-Son89** [comment]
> Also result looks different when is required or is Purchasable is enabled, Design: image PR: File

I discussed with Pedro from the design team, and the border color for the required icon should be able to be set externally. e.g so that certain groups can use it themed for their group.

Although, now I think about it. The purple used in the design should probably be the default color unless another border color is set.

**Comment by cammellos** [comment]
> > Also result looks different when is required or is Purchasable is enabled, Design: image PR: File
> 
> I discussed with Pedro from the design team, and the border color for the required icon should be able to be set externally. e.g so that certain groups can use it themed for their group.
> 
> Although, now I think about it. The purple used in the design should probably be the default color unless another border color is set.

Yes, I think they always come in purple (two different shades, one for light & one for dark), so no need to make the color customizable, as well because the icons at the moment they come already with that color:

https://www.figma.com/file/qLLuMLfpGxK9OfpIavwsmK/Mobile-Iconset?node-id=971%3A71

This one should be the right size, `main-icons/verified` is the old one, you can export also the dark mode version from figma (similarly for the + sign), let me know if you need any help and I can do that for you, if you'd like me to.

**Comment by J-Son89** [comment]
> > > Also result looks different when is required or is Purchasable is enabled, Design: image PR: File
> > 
> > 
> > I discussed with Pedro from the design team, and the border color for the required icon should be able to be set externally. e.g so that certain groups can use it themed for their group.
> > Although, now I think about it. The purple used in the design should probably be the default color unless another border color is set.
> 
> Yes, I think they always come in purple (two different shades, one for light & one for dark), so no need to make the color customizable, as well because the icons at the moment they come already with that color:
> 
> https://www.figma.com/file/qLLuMLfpGxK9OfpIavwsmK/Mobile-Iconset?node-id=971%3A71
> 
> This one should be the right size, `main-icons/verified` is the old one, you can export also the dark mode version from figma (similarly for the + sign), let me know if you need any help and I can do that for you, if you'd like me to.

 I removed border-color and color from the required and purchasable-icons so that these just come from the icons directly.
 
 also I added a default value for the required icon (the new icon above) and set the default value of the border-color of the whole tag to match too.
 I also added a prop so that the required icon can be set an the border-color of the whole tag too. This is so that other versions of the theme tag can be set. This is for the use case I discussed with Pedro from the design team.

File: src/quo2/screens/main.cljs
Comment (J-Son89) [reply src/quo2/screens/main.cljs]: done

**Comment by flexsurfer** [comment]
regarding icons naming https://discord.com/channels/624205794384281629/626744384565674034/991236399737098240

**Comment by J-Son89** [comment]
> regarding icons naming https://discord.com/channels/624205794384281629/626744384565674034/991236399737098240

updated the files I used to match this

File: src/quo2/foundations/colors.cljs
Comment (flexsurfer) [inline src/quo2/foundations/colors.cljs]: no Customisation?

Comment (J-Son89) [reply src/quo2/foundations/colors.cljs]: whoops, must remove that too! thanks

File: src/quo2/components/tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/tag.cljs]: If components are created from the bottom up, like small/base components first, and then used to create new components with the same padding/margin as Figma, then the height will automatically be 32. And if height is not hard-coded then border-width automatically will be applied outside and we don't need to calculate new height as for `border-width` (as same as you don't need to calculate new width). (no strong opinion, just thinking out loud)

File: src/quo2/components/tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/tag.cljs]: why 100?

File: src/quo2/components/tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/tag.cljs]: Thanks for adding comments.
From comment it looks like you meant to pass label as argument, but then you used `text-value` ?

File: src/quo2/components/token_tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/token_tag.cljs]: better to move `[rn/view ...` to new line. (When it is possible, please, keep lines' length shorter/equal 80 chars.)

File: src/quo2/components/tag.cljs
Comment (J-Son89) [reply src/quo2/components/tag.cljs]: hmm, I think I just maxed it out. I guess at some point it makes no odds so I will adjust it down to the right setting

File: src/quo2/components/tag.cljs
Comment (J-Son89) [reply src/quo2/components/tag.cljs]: will adjust!

File: src/quo2/components/token_tag.cljs
Comment (J-Son89) [reply src/quo2/components/token_tag.cljs]: noted, thanks Parvesh :)

**Comment by flexsurfer** [comment]
@jo-mut i see a tag component in you community PR , could please take a look at this PR

File: src/quo2/components/tag.cljs
Comment (J-Son89) [reply src/quo2/components/tag.cljs]: yeah so I just added another view around this so I could hardcode the height and with of the child view. Which means the border height will be added outside automatically.

File: src/quo2/components/token_tag.cljs
Comment (cammellos) [inline src/quo2/components/token_tag.cljs]: Indentation

**Comment by J-Son89** [comment]
> @jo-mut i see a tag component in you community PR , could please take a look at this PR

I'll be following onto John's work on communities in some other work so I can also look into making them both referencing the same base tag component

**Review by cammellos — APPROVED** [review]

**Comment by jo-mut** [comment]
I think all tags are sharing some properties. I implemented a base-tag but I think some properties are also missing on the base-tag. I will look into it

**Comment by J-Son89** [comment]
> I think all tags are sharing some properties. I implemented a base-tag but I think some properties are also missing on the base-tag. I will look into it

cool, we can follow up to refactor both of these once this work is merged in.

**Comment by jo-mut** [comment]
yeah its just refactoring. As it is looks okay

**Review by jo-mut — COMMENTED** [review]
I think this okay to merge.

File: src/quo2/components/tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/tag.cljs]: No strong opinion, but it will be great if text-value/label is passed as argument, so that all components follow same pattern
Ex. [1](https://github.com/status-im/status-mobile/blob/8e3258769e7b1e6b059d403ffe5337871635bea5/src/quo2/components/button.cljs#L109) [2](https://github.com/status-im/status-mobile/blob/8e3258769e7b1e6b059d403ffe5337871635bea5/src/quo2/components/counter.cljs#L22)

File: src/quo2/components/tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/tag.cljs:17]: Just a suggestion, but we usually indent values inside map (more readable). **[Example](https://github.com/status-im/status-mobile/blob/8e3258769e7b1e6b059d403ffe5337871635bea5/src/quo2/components/button.cljs#L161)**
image

**Review by Parveshdhull — APPROVED** [review]
Other than below comments, looks great. Thank you

(As you mentioned on discord, this PR already have approval from design team, So please feel free to merge it)

File: src/quo2/components/token_tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/token_tag.cljs]: If possible, please indent all maps

File: src/quo2/components/tag.cljs
Comment (J-Son89) [reply src/quo2/components/tag.cljs]: thanks Parvesh, I'll adjust 👍

File: src/quo2/components/tag.cljs
Comment (J-Son89) [reply src/quo2/components/tag.cljs:17]: cool :)

**Pull request merged**
